Program.cs

```diff
@@ -1,2 +1,189 @@
+
+                Console.Write("Type another number, and then press Enter: ");
+                double cleanNum2 = ValidateNumberInput(calculator);
+
+                DisplayOperatorOptions();
+
+                string op = ValidateOperatorInput();
+
+                double result = CalculateResult(calculator, cleanNum1, cleanNum2, op);
+                PrintResult(result);
+
+                DisplayCalculatorUsage(calculator);
+
+                DisplayMemory(calculator);
+
+                DisplayCleanupOptions();
+
+                endApp = ProcessCleanup(endApp, calculator);
+
+            }
+            calculator.Finish();
+            return;
+        }
+
+        // This seems to be an ugly solution. Where can I put the endApp bool?
+        private static bool ProcessCleanup(bool endApp, Calculator calculator)
+        {
+            string selection = Console.ReadLine();
+            switch (selection)
+            {
+                case "n":
+                    endApp = true;
+                    break;
+                case "c":
+                    calculator.ClearHistory();
+                    break;
+                default:
+                    break;
+            }
+
+            return endApp;
+        }
+
+        private static void DisplayCalculatorUsage(Calculator calculator)
+        {
+            Console.Write($"This calculator has been used {calculator.CalculationsCompleted} times.\n");
+        }
+
+        private static void PrintResult(double result)
+        {
+            if (double.IsNaN(result))
+            {
+                Console.WriteLine("This operation will result in a mathematical error.\n");
+            }
+            else Console.WriteLine("Your result: {0:0.##}\n", result);
+        }
+
+        private static double CalculateResult(Calculator calculator, double cleanNum1, double cleanNum2, string op)
+        {
+            double result = 0;
+            try
+            {
+                result = calculator.DoOperation(cleanNum1, cleanNum2, op);
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine("Oh no! An exception occurred trying to do the math.\n - Details: " + e.Message);
+            }
+            return result;
+        }
+
+        private static void DisplayMemory(Calculator calculator)
+        {
+            Console.WriteLine("---------Memory--------\n");
+            string memory = string.Empty;
+            int i = 0;
+            foreach (Calculation calc in calculator.CalculationMemory)
+            {
+                memory += $"{i}: {calc}";
+                i++;
+            }
+            Console.WriteLine(memory);
+            Console.WriteLine("To reference memory in your input type 'M' followed by the memory reference (e.g. M2)");
+        }
+
+        private static void DisplayOperatorOptions()
+        {
+            Console.WriteLine("Choose an operator from the following list:");
```

Comment:
You can write Console.WriteLine(@"my multiline string 
                                                          second line 
                                                          third line")

This way you don't need so many Console.WriteLine statements.

Reply:
I could not find any particularly neater way of expressing a string list using the verbatim symbol.